src/api/api.ts

```diff
@@ -1,65 +1,68 @@
-import { JsonRPCRequest, JsonRPCResponse, SUPPORTED_METHODS } from '@types';
+import { ipcMain, WebContents } from 'electron';
+
+import { IPC_CHANNELS, JsonRPCRequest, JsonRPCResponse, SUPPORTED_METHODS } from '@types';
 import { safeJSONParse } from '@utils';
 
 import { isValidRequest } from './validators';
 
+const toJsonRpcResponse = (response: Omit<JsonRPCResponse, 'jsonrpc'>) => {
+  return { jsonrpc: '2.0', ...response };
+};
+
+const requestSigning = (
+  request: JsonRPCRequest,
+  webContents: WebContents
+): Promise<JsonRPCResponse> => {
+  // @todo Cleaner way of doing this?
+  // @todo Reject?
+  return new Promise((resolve, _reject) => {
+    webContents.send(IPC_CHANNELS.API, request);
+
+    ipcMain.on(IPC_CHANNELS.API, (_event, arg) => {
+      const response = arg as JsonRPCResponse;
+      if (response.id === request.id) {
+        resolve(response);
+      }
+    });
+  });
+};
+
 // Replies follow: https://www.jsonrpc.org/specification
-export const handleRequest = (
+export const handleRequest = async (
   data: string,
-  sendToUI: (message: any) => void,
-  reply: (response: JsonRPCResponse) => void
-) => {
+  webContents: WebContents
+): Promise<JsonRPCResponse> => {
   // @todo: Further sanitation?
-  const json = safeJSONParse(data);
-  if (json[0] !== null) {
-    reply({
+  const [valid, parsed] = safeJSONParse(data);
+  if (valid !== null) {
+    return toJsonRpcResponse({
       id: null,
-      jsonrpc: '2.0',
       error: { code: '-32700', message: 'Parse error' }
     });
-    return;
   }
-  const request = json[1] as JsonRPCRequest;
+  const request = parsed as JsonRPCRequest;
   if (!Object.values(SUPPORTED_METHODS).includes(request.method)) {
```

Comment:
This is important.
Extract to helper function `isSupported` | `isAuthorised` | `isWhitelisted` and test.